### PR TITLE
Adds spell checker to the inspect tool for qbk and doxygen comments.

### PR DESCRIPTION
**Pull request by Bcorde5**
The initial test should fail the inspect check.  Feedback is necessary here in order for me to determine what to keep and what to delete in terms of spell checking.

### CODE REVIEW

File: tools/inspect/inspect.cpp [hunk 1]
```diff
@@ -1026,10 +1038,7 @@ void print_output(std::ostream& out, inspector_list const& inspectors)
       // we should not use a table, of course [gps]
       "<table>\n"
       "<tr>\n"
-      "<td>"
-      "<a href = \"https://github.com/STEllAR-GROUP/hpx\">"
-      "<img src=\"http://stellar.cct.lsu.edu/files/stellar100.png\" alt=\"STE||AR logo\" />"
-      "</a>\n"
+      "<td><img src=\"http://stellar.cct.lsu.edu/files/stellar100.png\" alt=\"STE||AR logo\" />"
```
Comment (hkaiser) [inline tools/inspect/inspect.cpp]: This change is probably unintended.

**Comment by hkaiser** [comment]
Where is the file `dictionary.hpp`? How is it built?

**Comment by Bcorde5** [comment]
The first mistake is fixed, but are you available to discuss the second sometime today?

**Comment by hkaiser** [comment]
From looking at https://circle-artifacts.com/gh/STEllAR-GROUP/hpx/915/artifacts/0/tmp/circle-artifacts.9uWSwYn/hpx_inspect_report.html we can see:

- some words miss the last character
- all doxygen commands (everything which starts with an `'\\'` can be ignored (i.e. `\param` or `\endcond`)
- single character 'words' can be ignored
- the dash character (`'-'`) should be treated as a delimiter as well

**Comment by Bcorde5** [comment]
should I add instructions for the python script?

**Comment by Bcorde5** [comment]
So a weird thing is happening, where the build forgets the last letter of some of the words, yet in my tests, the letters are added.  I believe this has something to do with the >= function in line 146 of the spell_check.cpp file.

**Comment by Bcorde5** [comment]
Currently looking for text files for names and locations, does anyone have hpx terms, or is there another option for it?

**Comment by sithhell** [comment]
Should the spellchecker issues be addressed in this PR or do we want to create a second one for it?

**Comment by hkaiser** [comment]
> Should the spellchecker issues be addressed in this PR or do we want to create a second one for it?

If we don't address it in this PR no builds will succeed. So yes, the spell fixes should be part of this PR.

**Comment by hkaiser** [comment]
Wedecided to close this for the time being.

**Pull request closed without merging**